src/java/test/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -888,4 +888,127 @@ public void testWithOnlyMinSessionTimeout() throws Exception {
                 maxSessionTimeOut, quorumPeer.getMaxSessionTimeout());
     }
 
+    @Test
+    public void testTxnAheadSnapInRetainDB() throws Exception {
```

Comment:
nit: I don't think we use the terminology "RetainDB" anywhere else. Perhaps we can get rid of "retain"?

Reply:
done